cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds

```diff
@@ -407,23 +407,67 @@ var __AdyenHelper: Object = {
         var shippingAddress = order.defaultShipment.shippingAddress;
         jsonObject["countryCode"] = shippingAddress.countryCode.value.toUpperCase();
 
+        var shippingStreet = "";
+        var shippingHouseNumberOrName = "";
+
+        if (!empty(shippingAddress.address1)) {
+            shippingStreet = shippingAddress.address1;
+
+            if (!empty(shippingAddress.address2)) {
+
+                shippingHouseNumberOrName = shippingAddress.address2;
+
+                if (paymentMethod.indexOf("afterpaytouch") > -1) {
+                    shippingHouseNumberOrName = "";
+                    shippingStreet += ' ' + shippingAddress.address2;
+                }
+
+            }
+
+        } else {
+            shippingStreet = 'N/A';
+        }
+
         jsonObject.deliveryAddress = {
             "city": (!empty(shippingAddress.city) ? shippingAddress.city : "N/A"),
             "country": (!empty(shippingAddress.countryCode) ? shippingAddress.countryCode.value.toUpperCase() : "ZZ"),
-            "houseNumberOrName": (!empty(shippingAddress.address2) ? shippingAddress.address2 : ""),
+            "houseNumberOrName": shippingHouseNumberOrName,
             "postalCode": (!empty(shippingAddress.postalCode) ? shippingAddress.postalCode : ""),
             "stateOrProvince": (!empty(shippingAddress.stateCode) ? shippingAddress.stateCode : "N/A"),
-            "street": (!empty(shippingAddress.address1) ? shippingAddress.address1 : "N/A")
+            "street": shippingStreet
         };
 
         var billingAddress = order.getBillingAddress();
+
+        var billingStreet = "";
+        var billingHouseNumberOrName = "";
+
+        if (!empty(billingAddress.address1)) {
+            billingStreet = billingAddress.address1;
+
+            if (!empty(billingAddress.address2)) {
+
+                billingHouseNumberOrName = billingAddress.address2;
+
+                if (paymentMethod.indexOf("afterpaytouch") > -1) {
+                    billingHouseNumberOrName = "";
+                    billingStreet += ' ' + billingAddress.address2;
+                }
+
+            }
+
+        } else {
+            billingStreet = 'N/A';
```

Comment:
lot of duplicate code. Can we move both to a separate function to have smaller methods with clear responsibility and as well re-use code. Let's add a unit test for this as well.